bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlSequence.java

```diff
@@ -637,4 +640,27 @@ private Type getSequenceType(Type tempExprType) {
                 return PredefinedTypes.TYPE_TEXT;
         }
     }
+
+    private void initializeIteratorNextReturnType() {
+        Type childrenType;
+        LinkedHashSet<Type> types = new LinkedHashSet<>();
```

Comment:
Why we use LinkedHashSet here?

Reply:
`LinkedHashSet` has been used because it will preserve the insertion order and it will define the iteration ordering.